File: snippets/lesson-3-transform.ts
import { range, Observable } from 'rxjs';
import { filter, map, take } from 'rxjs/operators';

const source$: Observable<number> = range(1, 8);

const processed$: Observable<number> = source$.pipe(
  filter((x: number) => x % 2 === 0),
  map((x: number) => x * 10),
  take(3),
);

processed$.subscribe((result: number) => console.log(result));

File: snippets/lesson-3-transform.cs
using System;
using System.Reactive.Linq;

IObservable<int> sourceStream = Observable.Range(1, 8);

IObservable<int> processedStream = sourceStream
    .Where((int x) => x % 2 == 0)
    .Select((int x) => x * 10)
    .Take(3);

processedStream.Subscribe((int result) => Console.WriteLine(result));